File: solutions/Virtual-Assistant/src/csharp/skills/todoskill/Dialogs/Cancel/CancelResponses.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using Microsoft.Bot.Builder.TemplateManager;
using ToDoSkill.Dialogs.Cancel.Resources;

namespace ToDoSkill
{
    public class CancelResponses : TemplateManager
    {
        // Constants
        public const string _cancelConfirmed = "Cancel.CancelConfirmed";

        // Fields
        private static LanguageTemplateDictionary _responseTemplates = new LanguageTemplateDictionary
        {
            ["default"] = new TemplateIdMap
            {
                { _cancelConfirmed, (context, data) => CancelStrings.CANCEL_CONFIRMED },
            },
            ["en"] = new TemplateIdMap { },
            ["fr"] = new TemplateIdMap { },
        };

        public CancelResponses()
        {
            Register(new DictionaryRenderer(_responseTemplates));
        }
    }
}

File: templates/Enterprise-Template/src/typescript/src/dialogs/cancel/cancelResponses.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License

import { DictionaryRenderer, LanguageTemplateDictionary } from "../templateManager/dictionaryRenderer";
import { TemplateManager } from "../templateManager/templateManager";
import * as i18n from "i18n";

export class CancelResponses extends TemplateManager {
    // Constants
    public static readonly _confirmPrompt: string = "Cancel.ConfirmCancelPrompt";
    public static readonly _cancelConfirmed: string = "Cancel.CancelConfirmed";
    public static readonly _cancelDenied: string = "Cancel.CancelDenied";

    // Fields
    private static readonly _responseTemplates: LanguageTemplateDictionary = new Map([
        ["default", new Map([
            [CancelResponses._confirmPrompt, () => Promise.resolve(i18n.__("cancel.prompt"))],
            [CancelResponses._cancelConfirmed, () => Promise.resolve(i18n.__("cancel.confirmed"))],
            [CancelResponses._cancelDenied, () => Promise.resolve(i18n.__("cancel.denied"))],
        ])]
    ]);

    constructor() {
        super();
        this.register(new DictionaryRenderer(CancelResponses._responseTemplates));
    }
}
